arangod/Cache/Manager.cpp

```diff
@@ -196,6 +204,12 @@ void Manager::shutdown() {
     if (!_shuttingDown) {
       _shuttingDown = true;
     }
+
+    while (globalProcessRunning()) {
+      // wait for rebalancer and migration tasks to complete
+      std::this_thread::yield();
```

Comment:
yield doesn't work like this at least on linux.

In general issue is scheduler algorithm which give every thread quant of time.
Now your thread executed -> so then you call sched_yield scheduler see that now time for your thread and your thread continue executing.

As Linus said using sched_yield almost always mistake (exception probably is not default scheduler something like real-time scheduler, fifo scheduler etc -- slow but with some guarantees)

cpu_relax than sleep should be used.
Example https://github.com/facebook/folly/blob/main/folly/synchronization/detail/Sleeper.h

Reply:
Ok, I added `cpu_relax()` here.